lib/active_merchant/billing/gateways/authorize_net.rb

```diff
@@ -180,20 +171,139 @@ def scrub(transcript)
+
+        commit(:cim_capture) do |xml|
+          add_order_id(xml, options)
+          xml.transaction do
+            xml.profileTransPriorAuthCapture do
+              xml.amount(amount(amount))
+              xml.transId(transaction_id)
+            end
+          end
+        end
+      end
+
+      def normal_capture(amount, authorization, options)
+        transaction_id = split_authorization(authorization)[0]
+
+        commit(:capture) do |xml|
+          add_order_id(xml, options)
+          xml.transactionRequest do
+            xml.transactionType('priorAuthCaptureTransaction')
+            xml.amount(amount(amount))
+            xml.refTransId(transaction_id)
+            add_invoice(xml, options)
+            add_user_fields(xml, amount, options)
+          end
+        end
+      end
+
+      def cim_refund(amount, authorization, options)
+        transaction_id, card_number = split_authorization(authorization)
+
+        commit(:cim_refund) do |xml|
+          add_order_id(xml, options)
+          xml.transaction do
+            xml.profileTransRefund do
+              xml.amount(amount(amount))
+              xml.creditCardNumberMasked(card_number)
+              add_invoice(xml, options)
+              xml.transId(transaction_id)
+            end
+          end
+        end
+      end
+
+      def normal_refund(amount, authorization, options)
+        transaction_id, card_number = split_authorization(authorization)
+
+        commit(:refund) do |xml|
+          xml.transactionRequest do
+            xml.transactionType('refundTransaction')
+            xml.amount(amount.nil? ? 0 : amount(amount))
+            xml.payment do
+              xml.creditCard do
+                xml.cardNumber(card_number || options[:card_number])
+                xml.expirationDate('XXXX')
+              end
+            end
+            xml.refTransId(transaction_id)
+
+            add_customer_data(xml, nil, options)
+            add_user_fields(xml, amount, options)
+          end
+        end
+      end
+
+      def cim_void(authorization, options)
+        commit(:cim_void) do |xml|
+          add_order_id(xml, options)
+          xml.transaction do
+            xml.profileTransVoid do
+              xml.transId(split_authorization(authorization)[0])
+            end
+          end
+        end
+      end
+
+      def normal_void(authorization, options)
+        commit(:void) do |xml|
+          add_order_id(xml, options)
+          xml.transactionRequest do
+            xml.transactionType('voidTransaction')
+            xml.refTransId(split_authorization(authorization)[0])
+            add_user_fields(xml, nil, options)
+          end
+        end
+      end
+
       def add_payment_source(xml, source)
         return unless source
-        if card_brand(source) == 'check'
+
+        case source
+        when String
+          add_token_payment_method(xml, source)
+        when Check
```

Comment:
`is_a?` checks are problematic, since it's common to define your own "quacks like" payment methods. Can you use a `respond_to?` instead?